sdk/daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -492,20 +501,54 @@ class Engine(val config: EngineConfig) {
             }
 
             val meta = Tx.Metadata(
-              submissionSeed = None,
+              submissionSeed = submissionInfo.map(_.submissionSeed),
               preparationTime = machine.preparationTime,
               usedPackages = deps,
               timeBoundaries = machine.getTimeBoundaries,
               nodeSeeds = nodeSeeds,
               globalKeyMapping = globalKeyMapping,
               disclosedEvents = disclosedCreateEvents,
             )
+
             config.profileDir.foreach { dir =>
               val desc = Engine.profileDesc(tx)
               val profileFile = dir.resolve(s"${meta.preparationTime}-$desc.json")
               machine.profile.name = s"${meta.preparationTime}-$desc"
               machine.profile.writeSpeedscopeJson(profileFile)
             }
+            config.snapshotDir.zip(submissionInfo).foreach {
```

Comment:
When `config.snapshotDir` and `submissionInfo` are non-empty, then we save the submissions transaction tree result to the snapshot directory (on a per participant basis).